internal/services/automation/automation_job_schedule_resource.go

```diff
@@ -232,7 +235,16 @@ func resourceAutomationJobScheduleDelete(d *pluginsdk.ResourceData, meta interfa
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := jobschedule.ParseJobScheduleID(d.Id())
+	tfID, err := parse.AutomationJobScheduleID(d.Id())
+	if err != nil {
+		return err
+	}
+	js, err := GetJobScheduleFromTFID(ctx, client, tfID)
+	if err != nil || js == nil {
```

Comment:
If `js == nil`, shoudl we just return `nil` as it indicates the schedule doesn't exist?

Reply:
I men can we split this to two `if` to make the intent clear (i.e. to avoid being smart)?

```
if err != nil {
  return err
}

if js == nil {
  return nil
}
```